src/function/algebra/polynomialRoot.js

```diff
@@ -0,0 +1,152 @@
+import { factory } from '../../utils/factory.js'
+
+const name = 'polynomialRoot'
+const dependencies = [
+  'typed',
+  'isZero',
+  'equalScalar',
+  'add',
+  'subtract',
+  'multiply',
+  'divide',
+  'sqrt',
+  'unaryMinus',
+  'cbrt',
+  'typeOf',
+  'im',
+  're'
+]
+
+export const createPolynomialRoot = /* #__PURE__ */ factory(name, dependencies, ({
+  typed,
+  isZero,
+  equalScalar,
+  add,
+  subtract,
+  multiply,
+  divide,
+  sqrt,
+  unaryMinus,
+  cbrt,
+  typeOf,
+  im,
+  re
+}) => {
+  /**
+   * Finds the numerical values of the distinct roots of a polynomial with real or complex coefficients.
+   * Currently operates only on linear, quadratic, and cubic polynomials using the standard
+   * formulas for the roots.
+   *
+   * Syntax:
+   *
+   *     polynomialRoot(constant, linearCoeff, quadraticCoeff, cubicCoeff)
+   *
+   * Examples:
+   *     // linear
+   *     math.polynomialRoot(6, 3)                                        // [-2]
+   *     math.polynomialRoot(math.complex(6,3), 3)                        // [-2 - i]
+   *     math.polynomialRoot(math.complex(6,3), math.complex(2,1))        // [-3 + 0i]
+   *     // quadratic
+   *     math.polynomialRoot(2, -3, 1)                                    // [2, 1]
+   *     math.polynomialRoot(8, 8, 2)                                     // [-2]
+   *     math.polynomialRoot(-2, 0, 1)                                    // [1.4142135623730951, -1.4142135623730951]
+   *     math.polynomialRoot(2, -2, 1)                                    // [1 + i, 1 - i]
+   *     math.polynomialRoot(math.complex(1,3), math.complex(-3, -2), 1)  // [2 + i, 1 + i]
+   *     // cubic
+   *     math.polynomialRoot(-6, 11, -6, 1)                               // [1, 3, 2]
+   *     math.polynomialRoot(-8, 0, 0, 1)                                 // [-1 - 1.7320508075688774i, 2, -1 + 1.7320508075688774i]
+   *     math.polynomialRoot(0, 8, 8, 2)                                  // [0, -2]
+   *     math.polynomialRoot(1, 1, 1, 1)                                  // [-1 + 0i, 0 - i, 0 + i]
+   *
+   * See also:
+   *     cbrt, sqrt
+   *
+   * @param {... number | Complex} coeffs
+   *     The coefficients of the polynomial, starting with with the constant coefficent, followed
+   *     by the linear coefficient and subsequent coefficients of increasing powers.
+   * @return {Array} The distinct roots of the polynomial
+   */
+
+  return typed(name, {
+    'number|Complex, ...number|Complex': (constant, restCoeffs) => {
```

Comment:
Does the function support BigNumbers? If not, is there a special reason for that?

Reply:
No, and the reason is 100% because right now there is no Complex<BigNumber>. Writing the code to switch down to number accuracy on BigNumber input when one or more of the roots happens to be complex would be a real pain, and I am just not sure how useful the function would be if it worked that way...

Reply:
(The point is right now the code is written not to care whether the answers come out complex or not, except for the last check to revert them to real if the imaginary part is negligible compared to the real. BigNumber support would make that impossible, especially as the cubic method sometimes uses complex intermediaries to end up with real values.)

Reply:
Ok thanks, that makes sense.